test/missiles_test.cpp

```diff
@@ -1,9 +1,20 @@
 #include <gtest/gtest.h>
 
+#include "engine/random.hpp"
 #include "missiles.h"
 
 using namespace devilution;
 
+TEST(Missiles, RotateBlockedMissileArrow)
+{
+	SetRndSeed(0);
+	Player &player = Players[0];
+	auto *missile = AddMissile({ 0, 0 }, { 0, 0 }, Direction::South, MIS_ARROW, TARGET_MONSTERS, player.getId(), 0, 0);
+	EXPECT_EQ(missile->_miAnimFrame, 1);
+	TestRotateBlockedMissile(*missile);
+	EXPECT_EQ(missile->_miAnimFrame, 16);
+}

```

Comment:
The name of this test is fine in that it only tests the case where `animType == MFILE_ARROWS`, however expanding the tests to also capture overflow and a regular non-wrapping rotation would be good.

Also should try structure tests so the broad setup happens first, then test setup immediately before the relevant call.

```suggestion
TEST(Missiles, RotateBlockedMissileArrow)
{
	SetRndSeed(0);
	Player &player = Players[0];
	// missile can be a copy or a reference, there's no nullptr check and the functions that use it don't expect the instance to be part of a global structure so it doesn't really matter for this use.
	Missile missile = *AddMissile({ 0, 0 }, { 0, 0 }, Direction::South, MIS_ARROW, TARGET_MONSTERS, player.getId(), 0, 0);
	EXPECT_EQ(missile._miAnimFrame, 1);
	
	SetRndSeed(0);
	TestRotateBlockedMissile(missile);
	EXPECT_EQ(missile._miAnimFrame, 16);
	
	SetRndSeed(/*find a seed which rotates the other direction*/);
	TestRotateBlockedMissile(missile);
	EXPECT_EQ(missile._miAnimFrame, 1);
	
	missile._miAnimFrame = 5;
	SetRndSeed(1234); // no idea what rotation this actually performs, might need to change the expectation to 4 instead of 6
	TestRotateBlockedMissile(missile);
	EXPECT_EQ(missile._miAnimFrame, 6);
}
```